### PR TITLE
Disable Babel error recovery for invalid escaped characters in identifiers

### LINKED ISSUE #8917: Unescape `Identifier` is not safe
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAdbBXYAJgAYBfTAXkwEZspIoMd9gBmYj8q6kAGhAgAHGAEt0yUAEMATtIgB3AAoyEaZCEkAbeZICea-gCNpksAGs4MAMqDTIqAHNkMaXjj8AFjAC2mgOoeIvBotmBwVqpBIgBuQbrqYGgGIPZocNIwiiYO3pLIAGZaafwAVmgAHgBCJuaWVpLecAAy9nAFRe4gZeVW9g6acACKeBDw7ZrFILbSadLqhpKGcJp8U9L2MH4iACYwHsgAHMT8gnJpfiaC6qdws9Ft-ACOI-BZQmookmgAtFBwcNsAatpHBniIQVlJDk8khChNOmlvCJnK4EX0BsNRm1YR1+DBFltdvskIQ8SYRJo+gBhCDeXLqW4AVlWeDSABVFh84ZNom4AJJQQGwKxgdbCACCgqsMF0A3GaVIpCAA)
```sh
--parser babel
```

**Input:**
```jsx
const \u{20} = 1
const \u{3000} = 1
```

**Output:**
```jsx
const   = 1;
const 　 = 1;

```

**Second Output:**
```jsx
SyntaxError: Unexpected token (1:9)
> 1 | const   = 1;
    |         ^
  2 | const 　 = 1;
  3 | 
```

**Expected behavior:**

Maybe we should keep it as it is, or at least check is it safe to unscape.

Another case 

```js
export asyn\\u{63} from "async";
```

https://github.com/prettier/prettier/pull/8914#discussion_r465624191

### PR DESCRIPTION
**Pull request by thorn0**
## Description

Fixes #8917

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parser-babel.js [hunk 1]
```diff
@@ -167,10 +167,15 @@ const messagesShouldThrow = new Set([
...
   "Unexpected token ','",
+  // ErrorMessages.EscapedCharNotAnIdentifier
+  "Invalid Unicode escape",
+  // ErrorMessages.MissingUnicodeEscape
+  "Expecting Unicode escape sequence \\uXXXX",
 ]);
 
 function shouldRethrowRecoveredError(error) {
   const [, message] = error.message.match(/(.*?)\s*\(\d+:\d+\)/);
+  console.info(message);
```
Comment (fisker) [inline src/language-js/parser-babel.js:178]: Remove this?

**Review by fisker — APPROVED** [review]
LGTM,  one  comment.

**Review by alexander-akait — APPROVED** [review]

**Comment by fisker** [comment]
Should we add [`no-console`](https://eslint.org/docs/rules/no-console) rule ?

**Comment by thorn0** [comment]
Good idea

**Comment by fisker** [comment]
Why we already have this rule overided here? https://github.com/prettier/prettier/blob/a322cf152a5f20c60892e2357f7e56858e2a2191/.eslintrc.yml#L87

**Comment by thorn0** [comment]
For the release script I guess

>

**Comment by fisker** [comment]
But we haven't enable it 😄

**Comment by thorn0** [comment]
https://eslint.org/docs/user-guide/migrating-to-6.0.0#eslintrecommended-has-been-updated

>

**Comment by thorn0** [comment]
It'll be best if we somehow enable it only for CI. I'd prefer the editor not to highlight it as a problem.

**Comment by fisker** [comment]
Need turn eslintrc to js file.

**Pull request merged**